nsxt/resource_nsxt_policy_edge_cluster.go

```diff
@@ -0,0 +1,406 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/sites/enforcement_points"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	"github.com/vmware/terraform-provider-nsxt/nsxt/metadata"
+)
+
+var policyEdgeClusterSchema = map[string]*metadata.ExtendedSchema{
+	"nsx_id":       metadata.GetExtendedSchema(getNsxIDSchema()),
+	"path":         metadata.GetExtendedSchema(getPathSchema()),
+	"display_name": metadata.GetExtendedSchema(getDisplayNameSchema()),
+	"description":  metadata.GetExtendedSchema(getDescriptionSchema()),
+	"revision":     metadata.GetExtendedSchema(getRevisionSchema()),
+	"tag":          metadata.GetExtendedSchema(getTagsSchema()),
+	"site_path": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			Description:  "Path to the site this Host Transport Node belongs to",
+			Optional:     true,
+			ForceNew:     true,
+			Default:      defaultInfraSitePath,
+			ValidateFunc: validatePolicyPath(),
+		},
+		Metadata: metadata.Metadata{
+			SchemaType: "string",
+			Skip:       true,
+		},
+	},
+	"enforcement_point": {
+		Schema: schema.Schema{
+			Type:        schema.TypeString,
+			Description: "ID of the enforcement point this Host Transport Node belongs to",
+			Optional:    true,
+			ForceNew:    true,
+			Default:     "default",
+		},
+		Metadata: metadata.Metadata{
+			SchemaType: "string",
+			Skip:       true,
+		},
+	},
+	"inter_site_forwarding_enabled": {
+		Schema: schema.Schema{
+			Type:     schema.TypeBool,
+			Optional: true,
+			Computed: true,
+		},
+		Metadata: metadata.Metadata{
+			SdkFieldName: "InterSiteForwardingEnabled",
+		},
+	},
+	"edge_cluster_profile": {
+		Schema: schema.Schema{
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validatePolicyPath(),
+		},
+		Metadata: metadata.Metadata{
+			SchemaType:   "string",
+			SdkFieldName: "EdgeClusterProfile",
+		},
+	},
+	"allocation_rule": {
```

Comment:
If we are rewriting this part manually, maybe just offer the boolean flag? (I'm guessing multiple rules are not supported)

Reply:
My assumption was that if this complexity was added in the API, then they might had some other rule functionality further on, so I left it as is. If you don't expect any, we can replace this with a boolean.